File: tests/fixtures/13-validation/pass/file7.go
// stripe/buggy_client_b05.go — Missing required field in request struct.
package stripe

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
)

// BUG: Missing `currency` field, which is required per manifest
type CreateChargeRequest struct {
	Amount int64  `json:"amount"`
	Source string `json:"source"`
	// currency field is MISSING (required in manifest)
}

type Charge struct {
	ID       string `json:"id"`
	Amount   int64  `json:"amount"`
	Currency string `json:"currency"`
	Status   string `json:"status"`
	Created  int64  `json:"created"`
}

func (c *Client) CreateCharge(req CreateChargeRequest) (*Charge, error) {
	body, err := json.Marshal(req)
	if err != nil {
		return nil, fmt.Errorf("marshal request: %w", err)
	}

	httpReq, err := http.NewRequest("POST", c.baseURL+"/charges", bytes.NewReader(body))
	if err != nil {
		return nil, fmt.Errorf("create request: %w", err)
	}

	httpReq.Header.Set("Authorization", "Bearer "+c.apiKey)
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := c.httpClient.Do(httpReq)
	if err != nil {
		return nil, fmt.Errorf("execute request: %w", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("stripe API error: status %d", resp.StatusCode)
	}

	var charge Charge
	if err := json.NewDecoder(resp.Body).Decode(&charge); err != nil {
		return nil, fmt.Errorf("decode response: %w", err)
	}

	return &charge, nil
}

File: tests/fixtures/16-validation/pass/file7.py
"""Stripe model with incomplete response shape."""

import httpx
from pydantic import BaseModel


class ChargeModel(BaseModel):
    """Incomplete charge model missing livemode field."""
    id: str
    amount: int
    currency: str
    status: str
    # Missing: livemode field (bool) present in real API responses


class StripeClient:
    def __init__(self, api_key: str):
        self.client = httpx.Client(
            base_url="https://api.stripe.com",
            headers={"Authorization": f"Bearer {api_key}"},
        )

    def create_charge(self, amount: int, currency: str, source: str) -> ChargeModel:
        """Create charge with incomplete model."""
        try:
            resp = self.client.post(
                "/v1/charges",
                data={"amount": amount, "currency": currency, "source": source},
            )
            resp.raise_for_status()
            # Pydantic will fail if response contains unexpected "livemode" field
            # (unless using extra="allow")
            return ChargeModel(**resp.json())
        except httpx.HTTPStatusError as e:
            raise ValueError(f"Charge failed: {e}")
